### DISCUSSION TITLE
[BluetoothLE] Advertisements device name being cached?

### DISCUSSION THREAD

**Discussion by pdecostervgls** [shinyorg/shiny#769] [Q&A] [answered]
I am using Shiny in my project and I noticed a case where if the advertised device name for a BLE device is being changed while the app is running, subsequent advertisements received will show the old device name. Restarting the app fixes this.
Is the advertisement name being cached internally?

**Comment by aritchie** [comment, marked as answer]
If you mean localname on the actual scan result advertisements isn't cached by Shiny nor should it be cached by the OS.  Device name on the other hand is stored between scans on Shiny, but is sometimes also cached by the OS for a short period of time.

**Reply by pdecostervgls** [reply]
Thanks for the reply, I tried using the LocalName but it only got me the first 8 characters of my device name, so that is an issue. I sometimes see the behavior that `IPeripheral::Name` only shows the first 8 charachters, but that is more exception than rule.
